auth-request.lua

```diff
@@ -24,6 +27,26 @@
 
 local http = require("haproxy-lua-http")
 
+core.register_action("auth-request", { "http-req" }, function(txn, be, path)
+	auth_request(txn, be, path, "HEAD", ".*", "-", "-")
+end, 2)
+
+core.register_action("auth-intercept", { "http-req" }, function(txn, be, path, method, hdr_req, hdr_succeed, hdr_fail)
+	hdr_req = globToLuaPattern(hdr_req)
+	hdr_succeed = globToLuaPattern(hdr_succeed)
+	hdr_fail = globToLuaPattern(hdr_fail)
+	auth_request(txn, be, path, method, hdr_req, hdr_succeed, hdr_fail)
+end, 6)
+
+function globToLuaPattern(glob)
+	if glob == "-" then
+		return "-"
+	end
+	-- magic chars: '^', '$', '(', ')', '%', '.', '[', ']', '*', '+', '-', '?'
+	-- https://www.lua.org/manual/5.4/manual.html#6.4.1
+	return "^" .. glob:gsub("[%^%$%(%)%%%.%[%]%+%-]", "%%%1"):gsub("*", ".*"):gsub("?", ".") .. "$"
```

Comment:
This looks horrible when you are used to regular expressions in other programming languages. I trust that you built this correctly :grinning:

Reply:
good point, I'll add a few more tests to explore this gsub.